lib/trino-parquet/src/main/java/io/trino/parquet/reader/ColumnReaderFactory.java

```diff
@@ -125,7 +133,12 @@ public ColumnReader create(PrimitiveField field, AggregatedMemoryContext aggrega
         }
         if (DATE.equals(type) && primitiveType == INT32) {
             if (annotation == null || annotation instanceof DateLogicalTypeAnnotation) {
-                return createColumnReader(field, valueDecoders::getInt32Decoder, INT_ADAPTER, memoryContext);
+                if (shouldUseHybridCalendarForDate) {
+                    return createColumnReader(field, encoding -> getDateInt32Decoder(valueDecoders.getInt32Decoder(encoding)), INT_ADAPTER, memoryContext);
+                }
+                else {
+                    return createColumnReader(field, valueDecoders::getInt32Decoder, INT_ADAPTER, memoryContext);
+                }
```

Comment:
Can we implement it as a ColumnAdaptation ? In this case the Date/Timestamp conversion can be unified and we could use them for ORC as well.